client/app/scripts/superdesk/ui/ui.js

```diff
@@ -397,6 +397,69 @@ define([
         };
     }
 
+    DropdownFocus.$inject = ['keyboardManager'];
+    function DropdownFocus(keyboardManager) {
+        return {
+            link: function (scope, elem) {
+                elem.bind('click', function (event) {
+                    _.defer(function () {
+                        var keyboardOptions = {inputDisabled: false},
+                            inputField = elem.find('input[type="text"]'),
+                            buttonList = elem.find('button:not([disabled]):not(.dropdown-toggle)');
+
+                        /*
+                         * If input field exist, put focus on it,
+                         * otherwise put it on first button in list
+                         */
+                        if (inputField.length > 0) {
+                            inputField.focus();
+                        } else {
+                            buttonList[0].focus();
+                        }
+
+                        keyboardManager.push('up', function () {
+                            var prevElem = elem.find('button:focus').parent('li').prev().children('button'),
+                                categoryButton = elem.find('.levelup button');
+
+                            if (prevElem.length > 0) {
+                                prevElem.focus();
+                            } else {
+                                inputField.focus();
+                                categoryButton.focus();
+                            }
+                        }, keyboardOptions);
+
+                        keyboardManager.push('down', function () {

```

Comment:
I think it will also need to pop it at some point, otherwise this will leak memory